### PR TITLE
Add DocSearch UI & configuration

### PR DESCRIPTION
**Pull request by ScottRudiger**
To increase developer productivity, we're adding a search feature that will return results from across our developer site.

### CODE REVIEW

**Comment by ScottRudiger** [comment]
Screen Shot 2019-10-15 at 2 30 22 PM

Results are:
- https://developers.taxjar.com/api/reference/#post-create-a-customer
- https://developers.taxjar.com/api/reference/#request
- https://developers.taxjar.com/api/reference/#parameters
- https://developers.taxjar.com/api/reference/#response
- https://developers.taxjar.com/api/reference/#attributes

So we're linking to all sections on the endpoint in the API reference. Is this level of granularity OK or does it come off as redundant?

File: docsearch.json [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  "start_urls": ["https://developers.taxjar.com"],
+  "stop_urls": ["https://developers.taxjar.com/api/reference/\\?"],
+  "selectors": {
+    "lvl0": "h1",
+    "lvl1": "h2",
+    "lvl2": "h3",
+    "lvl3": "h4",
+    "lvl4": "h5",
+    "lvl5": "h6",
+    "text": "p, li"
```
Comment (fastdivision) [inline docsearch.json]: @ScottRudiger Let's add `td` or a way to search endpoint params, attributes, and categories.

File: source/layouts/layout.erb [hunk 2]
```diff
@@ -15,6 +15,10 @@
     <% end %>
     <%= stylesheet_link_tag "main" %>
     <%= yield_content :css %>
+    <link
```
Comment (fastdivision) [inline source/layouts/layout.erb]: @ScottRudiger I'd make this single line and place it above `<%= yield_content :css %>` (line 17).

File: source/layouts/layout.erb [hunk 3]
```diff
@@ -76,5 +80,22 @@
...
+    <script src="https://cdn.jsdelivr.net/npm/docsearch.js@2/dist/cdn/docsearch.min.js"></script>
+    <script>
+      docsearch({
+        appId: '7ZBSP80X0K',
+        apiKey: '2a9d77b7d7b611725a62fd9ffb522880',
+        indexName: 'developers.taxjar.com',
+        inputSelector: '#docsearch'
+      });
+    </script>
+    <script>
```
Comment (fastdivision) [inline source/layouts/layout.erb]: @ScottRudiger Same feedback here as taxjar-api-docs. Can we use one selector for both, e.g. `#docsearch, #docsearch-mobile`?

Comment (ScottRudiger) [reply source/layouts/layout.erb]: Unfortunately not. (See explanation in [taxjar-api-docs.](https://github.com/taxjar/taxjar-api-docs/pull/56#discussion_r335761987))

**Pull request merged**